futures-util/src/lib.rs

```diff
@@ -54,6 +53,85 @@ macro_rules! delegate_sink {
     }
 }
 
+// FIXME: currently async/await is only available with std
+if_std! {
+    use std::marker::Unpin;
+    use std::mem::PinMut;
+
+    #[doc(hidden)]
+    pub fn assert_unpin<T: Future + Unpin>(_: &T) {}
+
+    /// A macro which returns the result of polling a future once within the
+    /// current `async` context.
+    ///
+    /// This macro is only usable inside of `async` functions, closures, and blocks.
+    #[macro_export]
+    macro_rules! poll {
+        ($x:expr) => {
+            await!($crate::poll($x))
+        }
+    }
+
+    #[doc(hidden)]
+    pub fn poll<F: Future + Unpin>(future: F) -> impl Future<Output = Poll<F::Output>> {
+        PollOnce { future }
+    }
+
+    #[allow(missing_debug_implementations)]
+    struct PollOnce<F: Future + Unpin> {
+        future: F,
+    }
+
+    impl<F: Future + Unpin> Future for PollOnce<F> {
+        type Output = Poll<F::Output>;
+        fn poll(mut self: PinMut<Self>, cx: &mut task::Context) -> Poll<Self::Output> {
+            Poll::Ready(PinMut::new(&mut self.future).poll(cx))
+        }
+    }
+
+    /// A macro which yields to the event loop once.
+    /// This is similar to returning `Poll::Pending` from a `Future::poll` implementation.
+    /// If `pending!` is used, the current task should be scheduled to receive a wakeup
+    /// when it is ready to make progress.
+    ///
+    /// This macro is only usable inside of `async` functions, closures, and blocks.
+    #[macro_export]
+    macro_rules! pending {
+        () => {
+            await!($crate::pending_once())
+        }
+    }
+
+    #[doc(hidden)]
+    pub fn pending_once() -> impl Future<Output = ()> {
```

Comment:
I've discovered that the tests of the `futures` crate use something similar called `DelayFuture`. Only difference is that it lets you specify the output.